### PR TITLE
Initial Application API Go client

### PR DESCRIPTION
**Pull request by jgustie**
This is the rough in for the Application API Go client, I want to get it up rather than sit on it while it gets cleaned up. Basically, it mimics how the experiments API client works. I need to reconcile a bit around how "names" work because the old model does work as cleanly for the new model (where the name is part of the entity body); this same issue affects the way `Title` works so expect some clean up there as well.

### CODE REVIEW

File: pkg/api/applications/v2/activity.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	"time"
+
+	"github.com/thestormforge/optimize-go/pkg/api"
+)
+
+type ActivityFeed struct {
+	HomePageURL string         `json:"home_page_url,omitempty"`
+	FeedURL     string         `json:"feed_url,omitempty"`
+	NextURL     string         `json:"next_url,omitempty"`
+	Hubs        []ActivityHub  `json:"hubs,omitempty"`
```
Comment (parsenz) [inline pkg/api/applications/v2/activity.go:30]: Cool to see this included. Would be nice to implement this as well eventually in the application service.

**Review by parsenz — APPROVED** [review]
Nice work! This looks great. Cool to see the provision for subscriptions done as well.

File: pkg/api/applications/v2/application.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+type applicationName string
+
+func (n applicationName) Name() string   { return string(n) }
+func (n applicationName) String() string { return string(n) }
+
+type Application struct {
+	api.Metadata `json:"-"`
+	Name_        string        `json:"name"`
```
Comment (bradbeam) [inline pkg/api/applications/v2/application.go:35]: Why `Name_` and not just `Name`?

File: pkg/api/applications/v2/http.go [hunk 3]
```diff
@@ -0,0 +1,556 @@
...
+
+func (h *httpAPI) CreateApplication(ctx context.Context, app Application) (api.Metadata, error) {
+	u := h.client.URL(endpointApplications).String()
+
+	req, err := httpNewJSONRequest(http.MethodPost, u, app)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, body, err := h.client.Do(ctx, req)
```
Comment (bradbeam) [inline pkg/api/applications/v2/http.go:80]: This almost got me with the similarities to net/http client.Do. Was worried we were leaking body for `http.StatusCreated`, but in api.Client body is just `[]byte`

**Review by bradbeam — APPROVED** [review]

File: pkg/api/applications/v2/application.go (hunk 2 above)
Comment (jgustie) [reply pkg/api/applications/v2/application.go:35]: Right now there is `Name()` function so it conflicts, I have a follow on PR that will clean it up (spoiler: I'm eliminating the function and changing the field to `Name`).

File: pkg/api/applications/v2/http.go (hunk 3 above)
Comment (jgustie) [reply pkg/api/applications/v2/http.go:80]: I shamelessly took inspiration from the [Prometheus Go client](https://github.com/prometheus/client_golang/blob/master/api/client.go) many moons ago.

**Pull request merged**